lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3277,9 +3311,10 @@ ParseResult FIRCircuitParser::parseModule(CircuitOp circuit,
 
     // Parse the body of this module after all prototypes have been parsed. This
     // allows us to handle forward references correctly.
-
+    TargetSet targetSet;
     deferredModules.push_back({moduleOp, portLocs, getLexer().getCursor(),
-                               std::move(moduleTarget), indent});
+                               std::move(moduleTarget), indent,
+                               std::move(targetSet)});
```

Comment:
I think you can pass `TargetSet()` directly instead of `std::move(targetSet)`

Reply:
I can if I don't use the `BumpPtrAllocator`, but if I do, it's implicitly deleted copy constructor causes problems 😬:

```
In file included from ../lib/Dialect/FIRRTL/Import/FIRParser.cpp:14:
In file included from ../lib/Dialect/FIRRTL/Import/FIRAnnotations.h:17:
../llvm/llvm/include/llvm/ADT/StringMap.h:139:9: error: call to implicitly-deleted copy constructor of 'llvm::BumpPtrAllocatorImpl<llvm::MallocAllocator, 4096, 4096, 128>'
        Allocator(RHS.Allocator) {
        ^         ~~~~~~~~~~~~~
../llvm/mlir/include/mlir/Support/LLVM.h:60:39: note: in instantiation of member function 'llvm::StringMap<llvm::NoneType, llvm::BumpPtrAllocatorImpl<llvm::MallocAllocator, 4096, 4096, 128>>::StringMap' requested here
template <typename AllocatorTy> class StringSet;
                                      ^
../llvm/llvm/include/llvm/Support/Allocator.h:88:3: note: copy constructor is implicitly deleted because 'BumpPtrAllocatorImpl<llvm::MallocAllocator, 4096, 4096, 128>' has a user-declared move constructor
  BumpPtrAllocatorImpl(BumpPtrAllocatorImpl &&Old)
  ^
1 error generated.
```

I'll punt on this change for now.

Reply:
Sounds good.  It took me like 42 tries, but I was able to get it to go in b6510fcd.  *shakes head at C++*